src/components/file-browser/FileBrowserDirectories.test.ts

```diff
@@ -0,0 +1,244 @@
+  it('renders the directory name', () => {
+    // Act
+    render(h(FileBrowserDirectory, {
+      activeDescendant: 'node-0',
+      level: 0,
+      id: 'node-0',
+      path: 'path/to/directory/',
+      provider: mockFileBrowserProvider,
+      rootLabel: 'Workspace bucket',
+      selectedDirectory: '',
+      setActiveDescendant: () => {},
+      onSelectDirectory: jest.fn()
+    }))
+
+    // Assert
+    screen.getByText('directory')
+  })
+
+  it('it calls onSelectDirectory callback with path when directory name is clicked', async () => {
+    // Arrange
+    const user = userEvent.setup()
+
+    const onSelectDirectory = jest.fn()
+    render(h(FileBrowserDirectory, {
+      activeDescendant: 'node-0',
+      id: 'node-0',
+      level: 0,
+      path: 'path/to/directory/',
+      provider: mockFileBrowserProvider,
+      rootLabel: 'Workspace bucket',
+      selectedDirectory: '',
+      setActiveDescendant: () => {},
+      onSelectDirectory
+    }))
+
+    // Act
+    const link = screen.getByText('directory')
+    await user.click(link)
+
+    // Assert
+    expect(onSelectDirectory).toHaveBeenCalledWith('path/to/directory/')
+  })
+
+  it('fetches and renders directory contents when expanded', async () => {
+    // Arrange
+    const user = userEvent.setup()
+
+    const directories = [
+      {
+        path: 'path/to/directory/subdirectory1'
+      },
+      {
+        path: 'path/to/directory/subdirectory2'
+      },
+      {
+        path: 'path/to/directory/subdirectory3'
+      }
+    ]
+
+    const useDirectoriesInDirectoryResult: UseDirectoriesInDirectoryResult = {
+      state: { directories, status: 'Ready' },
+      hasNextPage: undefined,
+      loadNextPage: () => Promise.resolve(),
+      loadAllRemainingItems: () => Promise.resolve(),
+      reload: () => Promise.resolve()
+    }
+
+    asMockedFn(useDirectoriesInDirectory).mockReturnValue(useDirectoriesInDirectoryResult)
+
+    render(h(FileBrowserDirectory, {
+      activeDescendant: 'node-0',
+      id: 'node-0',
+      level: 0,
+      path: 'path/to/directory/',
+      provider: mockFileBrowserProvider,
+      rootLabel: 'Workspace bucket',
+      selectedDirectory: '',
+      setActiveDescendant: () => {},
+      onSelectDirectory: jest.fn()
+    }))
+
+    // Act
+    const contentsFetchedBeforeExpanding = asMockedFn(useDirectoriesInDirectory).mock.calls.length > 0
+    const toggle = screen.getByTestId('toggle-expanded')
+    await user.click(toggle)
+    const contentsFetchedAfterExpanding = asMockedFn(useDirectoriesInDirectory).mock.calls.length > 0
+
+    const subdirectoryList = screen.getByRole('group')
+    const renderedSubdirectories = Array.from(subdirectoryList.children).map(el => el.children[1].textContent)
+
+    // Assert
+    expect(contentsFetchedBeforeExpanding).toBe(false)
+    expect(contentsFetchedAfterExpanding).toBe(true)
+
+    expect(renderedSubdirectories).toEqual(['subdirectory1', 'subdirectory2', 'subdirectory3'])
```

Comment:
It might also be good to add in an axe check to verify aria attributes/ids. 

```
expect(await axe(container)).toHaveNoViolations()
```

Reply:
Added in https://github.com/DataBiosphere/terra-ui/pull/3469/commits/ca23519197059376718c5a5107f814af4b2ff00c.